generate/schemagen/gen/generator.go

```diff
@@ -138,6 +142,20 @@ func Generate(pkgRegistry PackageRegistry, linker Linker, client APIServer, opti
 			}
 		}
 
+		if objType != nil {
+			for _, method := range objType.MethodNames() {
+				m, err := objType.Method(method)
+				if err != nil {
+					continue
+				}
+				// Remove functions that only have a single struct parameter
+				// and no return value. These are facade methods intended to
+				// hide and discontinue use of the method.
+				if m.Params == structType && m.Result == nil {
```

Comment:
So this is 'how we happen to have implemented it several times' but the actual logic for what gets exposed is here:
https://github.com/juju/rpcreflect/blob/b4903b859171fbfa2ddc4e263b7a82361173c169/type.go#L292

I wonder if it would be better to just use rpcreflect.ObjTypeOf in this code, since that is the authoritative source for where that information resides. That way we are using the exact code path for what is exposed in the API to determine what is exposed in python-libjuju.

Thoughts?